ignite/src/test/java/org/hibernate/ogm/datastore/ignite/utils/IgniteTestConfigurationBuilder.java

```diff
@@ -41,10 +40,9 @@ private IgniteConfiguration createConfig() {
 		binaryConfiguration.setCompactFooter( false );		// it is necessary only for embedded collections (@ElementCollection)
 		config.setBinaryConfiguration( binaryConfiguration );
 		TransactionConfiguration transactionConfiguration = new TransactionConfiguration();
-		transactionConfiguration.setDefaultTxConcurrency( TransactionConcurrency.OPTIMISTIC );
+		transactionConfiguration.setDefaultTxConcurrency( TransactionConcurrency.PESSIMISTIC );
```

Comment:
Is this needed for this PR? Or there is some other reason to change it?

Reply:
This change seems to be here for assessment of the case, that no reordering of keys is done when they are passed to the Ignite in the actual "getTuples" method, as the "getAll" accepts set of keys, the set should maintain the same iteration order that the list of keys passed to the "getTuples" method, otherwise it might introduce another source of deadlock in case of PESSIMISTIC (upfront locking) concurrency level.

Reply:
PESSIMISTIC had been in code from start of development this dialect and was changed 10 weeks ago. By mistake or misunderstanding, I believe. I have never had problems with tests due to PESSIMISTIC mode. I only corrected this mistake.

Reply:
Tests works with OPTIMISTIC too. But I and @YevIgn faced tx hanging when we use PESSIMISTIC . It can be a problem of Apache Ignite ( tested versions are 1.9 and 2.1). Anyway OPTIMISTIC must be a default concurrency level because it most safe level.

Reply:
What do you mean "safe level"?  OPTIMISTIC means no locking at all. I think it isn't serious to run tests with this transaction level

Reply:
Strictly speaking the issues, we faced with PESSIMISTIC setting, seem to be mostly related to the fact that in our case there was extensive use of one-to-many relationships, where two or more concurrent transactions accessed elements in different order. Or at least that is how I understood the problem back then.

Reply:
> OPTIMISTIC means no locking at all

Well, in general. It means that during an update or write the db will check that the previous data is not changed. It's a valid level and useful in many situations. We also have test in OGM to check for optimistic locking. I'm not sure how Apache Ignite handle it but maybe it can implement the `OptimisticLockingAwareGridDialect` interface.

Reply:
> Strictly speaking the issues, we faced with PESSIMISTIC setting, seem to be mostly related to the fact that in our case there was extensive use of one-to-many relationships, where two or more concurrent transactions accessed elements in different order. Or at least that is how I understood the problem back then.

Maybe too many fetch EAGER loading too many relationships and hitting too many entities?

Reply:
>OPTIMISTIC means no locking at all

In case of Apache Ignite OPTIMISTIC means locks acquisition on prepare step. Version check is performed on SERIALIZABLE isolation level. Other combinations of concurrency/isolation levels are subject to deadlock detection mechanism - https://apacheignite.readme.io/docs/transactions#section-deadlock-detection

Reply:
> Maybe too many fetch EAGER loading too many relationships and hitting too many entities?

It was certainly a deadlock, as the number of entities was fairly small - namely no more than a dozen, and it was actually quite simple to trace it to the different order of entries access.